pico/sbrk.c

```diff
@@ -0,0 +1,34 @@
+#include <sys/types.h>
+
+/* brk is handled entirely within the C library.  This limits METAL programs
+ * that use the C library to be disallowed from dynamically allocating memory
+ * without talking to the C library, but that sounds like a sane way to go
+ * about it.  Note that there is no error checking anywhere in this file, users
+ * will simply get the relevant error when actually trying to use the memory
+ * that's been allocated. */
+extern char metal_segment_heap_target_start;
+extern char metal_segment_heap_target_end;
+static char *_brk = &metal_segment_heap_target_start;
```

Comment:
The only difference I see between this an gloss/sys_sbrk.c is the underscore in `_brk`. Can we combine these into the same implementation?

Reply:
I think its a short enough piece of code that duplicating it will end up being less hassle than managing #ifdef's in the code? I've added the #ifdef's and pushed out the change; does that look reasonable?